presto-kinesis/src/main/java/io/prestosql/plugin/kinesis/s3config/S3TableConfigClient.java

```diff
@@ -0,0 +1,265 @@
+        ImmutableMap.Builder<SchemaTableName, KinesisStreamDescription> builder = ImmutableMap.builder();
+        for (KinesisStreamDescription stream : streamValues) {
+            builder.put(new SchemaTableName(stream.getSchemaName(), stream.getTableName()), stream);
+        }
+        return builder.build();
+    }
+
+    /**
+     * Shutdown any periodic update jobs.
+     */
+    @Override
+    public void run()
+    {
+        if (isUsingS3() && updateTaskHandle != null) {
+            updateTaskHandle.cancel(true);
+        }
+        return;
+    }
+
+    @PostConstruct
+    protected void startS3Updates()
+    {
+        if (this.bucketUrl.isPresent()) {
+            ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
+            this.updateTaskHandle =
+                    scheduler.scheduleAtFixedRate(() -> updateTablesFromS3(), 5, 600, TimeUnit.SECONDS);
+        }
+    }
+
+    /**
+     * Call S3 to get the most recent object list.
+     * <p>
+     * This is an object list request to AWS in the given "directory".
+     *
+     * @return
+     */
+    protected List<S3ObjectSummary> getObjectSummaries()
+    {
+        AmazonS3Client s3client = this.clientManager.getS3Client();
+        AmazonS3URI directoryURI = new AmazonS3URI(this.bucketUrl.get());
+
+        List<S3ObjectSummary> returnList = new ArrayList<S3ObjectSummary>();
+        try {
+            log.info("Getting the listing of objects in the S3 table config directory: bucket %s prefix %s :", directoryURI.getBucket(), directoryURI.getKey());
+            ListObjectsRequest req = new ListObjectsRequest().withBucketName(directoryURI.getBucket())
+                    .withPrefix(directoryURI.getKey() + "/").withDelimiter("/").withMaxKeys(25);
+            ObjectListing result;
+
+            do {
+                result = s3client.listObjects(req);
+
+                returnList.addAll(result.getObjectSummaries());
+                req.setMarker(result.getNextMarker());
+            }
+            while (result.isTruncated());
+
+            log.info("Completed getting S3 object listing.");
+        }
+        catch (AmazonServiceException ase) {
+            StringBuilder sb = new StringBuilder();
+            sb.append("Caught an AmazonServiceException, which means your request made it ");
+            sb.append("to Amazon S3, but was rejected with an error response for some reason.\n");
+            sb.append("Error Message:    " + ase.getMessage());
+            sb.append("HTTP Status Code: " + ase.getStatusCode());
+            sb.append("AWS Error Code:   " + ase.getErrorCode());
+            sb.append("Error Type:       " + ase.getErrorType());
+            sb.append("Request ID:       " + ase.getRequestId());
+            log.error(sb.toString(), ase);
+        }
+        catch (AmazonClientException ace) {
+            StringBuilder sb = new StringBuilder();
+            sb.append("Caught an AmazonClientException, " +
```

Comment:
We don't normally put these types of messages in exceptions. This seems more appropriate for documentation.